gradio/routes.py

```diff
@@ -571,8 +570,8 @@ async def predict(
                     content={"error": str(error) if show_error else None},
                     status_code=500,
                 )
-            root_path = app.get_blocks().config.get("root", "")
-            output = add_root_url(output, route_utils.strip_url(root_path))
+            root_path = route_utils.get_root_url(request)[: -len(f"/api/{api_name}")]
```

Comment:
Annoying that I have to do this manually, but if you use e.g. `request.url.path`, it also includes the root url if gradio is mounted within another fastapi app, which we don't want. So I think the most reliable solution is to just remove the suffix manually.